airflow/models.py

```diff
@@ -3367,6 +3367,36 @@ def val(cls):
 
     @classmethod
     @provide_session
+    def setdefault(cls, key, default, deserialize_json=False, session=None):
+        """
+        Like a Python builtin dict object, setdefault returns the current value
+        for a key, and if it isn't there, stores the default value and returns it.
+
+        :param key: Dict key for this Variable
+        :type key: String
+        :param: default: Default value to set and return if the variable
+        isn't already in the DB
+        :type: default: Mixed
+        :param: deserialize_json: Store this as a JSON encoded value in the DB
+         and un-encode it when retrieving a value
+        :return: Mixed
+        """
+        obj = session.query(cls).filter(cls.key == key).first()
```

Comment:
In that case it would be safer to use a proper sentinel instead of an empty dict:

```python
default_sentinel = object()
obj = Variable.get(key, default_var=default_sentinel)  # deserialize_json=False is the default, right?
if obj is default_sentinel:
    ...
```

Though, at this point, maybe it's better to ask forgiveness rather than permission:

```python
try:
    obj = Variable.get(key, deserialize_json=deserialize_json)
except ValueError:
    Variable.set(key, default_var, serialize_json=deserialize_json)
    return default_var
```

(as an aside, looking at this says to me that `Variable.get` should be raising a `KeyError` instead of a `ValueError`)